### ISSUE TITLE
Support custom documentation for setters

### ISSUE THREAD

**Issue by mre** [colin-kiegel/rust-derive-builder#13]
Consider this builder:
```rust
custom_derive! {
    #[derive(Debug, Default, Builder)]
    struct Channel {
        //! This is a comment for token
        token: i32,
        special_info: i32
    }
}
```
It would be great if there was a way to add the comments as documentation for the fields:

```rust
//! This is a comment for token
fn token<VALUE: Into<i32>>(mut self, value: VALUE) -> Self {
    self.token = value.into();
    self
}
```

No clue if that is possible, though. :smile:

**Comment by killercup** [comment]
This should be possible, `/// hi there` comments become `#[doc(hi there)]` attributes.

Would you want to differentiate between comments for the struct field and comments for the setter? (If the field is not public, its docs wouldn't be rendered, so it might be cool to just add them to the setter fn.)

**Comment by mre** [comment]
Thanks for the quick response.
Documentation for the setter would be enough for me right now. But a fallback for non-public fields would be totally awesome. Depends on how tricky it is to implement.

**Comment by colin-kiegel** [comment]
@mre thank you very much for your feedback. :-)

A simple solution would be to allow item doc-comments, but to skip over them and use auto-generated documentation for setters. Spontaneously that is what I would try first and then wait and see if there is a big need for something more sophisticated or magic.

**Comment by colin-kiegel** [comment]
Ok, for now we have auto-generated doc-comments (https://github.com/colin-kiegel/rust-derive-builder/commit/95f3b4b0a74a42a24393c4fd93cf53afe9fbddcf)
```rust
/// Set `$field_name`
///
/// auto-generated by [derive_builder](https://crates.io/crates/derive_builder)
```

Skipping doc-comments (and `pub` keywords) in the struct definition is a bit more delicate and therefore a pull request https://github.com/colin-kiegel/rust-derive-builder/pull/14

I am open for future changes - but I think this is at least an improvement of the current state. I will release now without the pull request. ;-)

**Comment by colin-kiegel** [comment]
implemented and published with [v0.2.0 (Changelog)](https://github.com/colin-kiegel/rust-derive-builder/blob/master/CHANGELOG.md#020---2016-08-22)
